### PR TITLE
Ft login api endpoint 160718913

### PR DESCRIPTION
**Pull request by idmega2000**
#### What does this PR do?
- it add a feature that allow registered user to be able to login into the app
#### Description of Task to be completed?
- [x] create login model
- [X] write test for login route

#### How should this be manually tested?
- clone repo and run npm install etc
- rename the .envsample to .env and fill the details
- use postman to input user data

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
- story type - feature
- story title - Users should be able to Login with login credentials
- story id - 160718913
#### Relevant Pivotal tracker story link
[login api endpoint link](https://www.pivotaltracker.com/story/show/160718913)

### CODE REVIEW

File: server/controllers/UserAuth.js [hunk 1]
```diff
@@ -24,5 +25,31 @@ class UserAuth {
...
+     * This function register a new user
+     * @param {object} req - the request object.
+     * @param {object} res - The response object.
+     * @returns {object} Returns success message and jwt token.
+     */
+  loginUsers(req, res) {
+    authModel.userSignIn(req.body)
+      .then((result) => {
+        if (result.rowCount === 0) {
+          return res.status(404).json({ error: 'User does not exist!' });
```
Comment (dogoyaro) [inline server/controllers/UserAuth.js:39]: I don't think this is the right status code. Please confirm.

File: server/test/userAuth.spec.js [hunk 2]
```diff
@@ -64,147 +64,146 @@ const emptyPassword = {
...
-            assert.equal(res.statusCode, 409);
-            assert.equal(res.body.error, 'User already exist');
-            done();
-          });
-      });
-
+  it('should return error when given empty useremail',
+    (done) => {
+      request.post(`${path}/signup`)
+        .send(worongInputType)
```
Comment (dogoyaro) [inline server/test/userAuth.spec.js:70]: Typo here.

File: server/controllers/UserAuth.js (hunk 1 above)
Comment (idmega2000) [reply server/controllers/UserAuth.js:39]: ok. I would confirm and get back to you

File: server/test/userAuth.spec.js (hunk 2 above)
Comment (idmega2000) [reply server/test/userAuth.spec.js:70]: thanks i have resolved this

**Pull request merged**